### PR TITLE
[WIP] Add readiness probe to certsyncer

### PR DESCRIPTION
**Pull request by tnozicka**
Certsyncer is a controller that "never" fails and installer pod places the initial certs there so we won't know until rotation happens and this break which is not covered in CI yet.

This will at least make sure the certsyncer caches are synced or we'd see it in the container status.

Next step is to wire the controller to go degraded when the container is not ready.

Requires:
 - [ ] Real bump of https://github.com/openshift/library-go/pull/601 (when merged)

/cc @deads2k @openshift/sig-master

### CODE REVIEW

**Comment by tnozicka** [comment]
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/662/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/3008/artifacts/e2e-aws/must-gather/registry-svc-ci-openshift-org-ci-op-lwp8pzd2-stable-sha256-64c63eedf863406fbc6c7515026f909a7221472cf70283708fb7010dd5e6139e/namespaces/openshift-kube-apiserver/pods/kube-apiserver-ip-10-0-133-197.ec2.internal/kube-apiserver-cert-syncer-6/kube-apiserver-cert-syncer-6/logs/current.log
```
2019-11-22T13:00:55.317438531Z I1122 13:00:55.317385       1 certsync_controller.go:301] CertSyncer caches synced
2019-11-22T13:00:55.317776101Z I1122 13:00:55.317744       1 certsync_controller.go:310] CertSyncer caches synced file "/var/run/informer-caches-synced" created.
```

**Comment by tnozicka** [comment]
cert-syncer doesn't hot loop, seems ok
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/662/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws-operator/1510/artifacts/e2e-aws-operator/must-gather/registry-svc-ci-openshift-org-ci-op-lwp8pzd2-stable-sha256-64c63eedf863406fbc6c7515026f909a7221472cf70283708fb7010dd5e6139e/namespaces/openshift-kube-apiserver/pods/kube-apiserver-ip-10-0-132-27.ec2.internal/kube-apiserver-cert-syncer-8/kube-apiserver-cert-syncer-8/logs/current.log

File: bindata/v4.1.0/kube-apiserver/pod.yaml [hunk 1]
```diff
@@ -100,6 +100,8 @@ spec:
       - --namespace=$(POD_NAMESPACE)
       - --destination-dir=/etc/kubernetes/static-pod-certs
       - --tls-server-name-override=localhost-recovery
+      - --caches-synced-file=/var/run/informer-caches-synced
+      - -v=2
```
Comment (sttts) [inline bindata/v4.1.0/kube-apiserver/pod.yaml]: why 2 not 4?

Comment (tnozicka) [reply bindata/v4.1.0/kube-apiserver/pod.yaml]: the default loglevel is historically set to 2, did we change it in 4.x?

**Pull request closed without merging**